### ISSUE TITLE
Move loss function to GPU

### ISSUE THREAD

**Issue by littleflow3r** [bentrevett/pytorch-sentiment-analysis#18]
Hi,
Could you please explain the intuition why we need to move the loss function to GPU?
`criterion = criterion.to(device)`

I almost never saw it in others.

Thanks for the great tutorial. I found your explanation quite intuitive.

**Comment by bentrevett** [comment]
I first saw it somewhere, but not sure where.

According to:
https://discuss.pytorch.org/t/if-the-criterion-is-not-set-to-cuda-but-the-network-is-set-to-cuda-does-the-operation-work-in-cuda/13290
https://discuss.pytorch.org/t/move-the-loss-function-to-gpu/20060/3

It only needs to be done if the criterion used has some internal parameters of its own. The criterion we use in these tutorials doesn't have any, but I don't see the harm in doing it anyway.

**Comment by littleflow3r** [comment]
Thanks!

**Comment by littleflow3r** [comment]
Hey, I am sorry to comment on this (supposedly) closed issue, but I have question related to moving things to GPU. 
From my understanding, we have to move both the model and data (any input batches/ tensor) to GPU. But in your code, I can only see `model = model.to(device)`
Does using this code already means that our data is also at GPU?
Or should we also explicitly move the data to GPU during training iteration, something like:
`batch.text = batch.text.to(device)` ?

Thanks, and sorry. I am at loss about making sure that pytorch using GPU to compute.

**Comment by bentrevett** [comment]
Yep, both the model and the data need to be on the same device. The reason we don’t need to explicitly put the data on the GPU is that it is already handled for us in the iterator, this is why we pass a ‘device’ to the ‘BucketIterator’, which will place the data on the GPU before yielding a batch of data.

**Comment by littleflow3r** [comment]
Oh! I forgot that we set the device in the iterator!
Thanks!
